### PR TITLE
Add exclude schemas

### PR DESCRIPTION
**Pull request by bplunkett-stripe**
[//]: # (README: Ensure you've read the CONTRIBUTING.MD and that your commits are signed)

### Description
[//]: # (A clear and concise description of the purpose of this Pull Request. What is being changed? Include any relevant background for this change.)
When using temp db factories (which have metadata tables), we need to tell the schema fetcher to ignore those tables.

This also technically switches acceptance tests to fetch non-public schemas; however, none of our tests actually have non-public schemas. 

The final step for supporting non-public schemas at this point is validating everything is actually working through the acceptance tests, updating the documentation, and removing the gate.
### Motivation
[//]: # (Why you made these changes. Link to any relevant issues.)
https://github.com/stripe/pg-schema-diff/issues/94

### Testing
[//]: # (Describe how you tested these changes)
Acceptance tests

### CODE REVIEW

File: internal/schema/schema.go [hunk 1]
```diff
@@ -382,34 +392,77 @@ func GetSchema(ctx context.Context, db queries.DBTX, opts ...GetSchemaOpt) (Sche
...
-func buildNameFilter(options getSchemaOptions) nameFilter {
-	if len(options.includeSchemas) == 0 {
+func buildNameFilter(options getSchemaOptions) (nameFilter, error) {
+	if intersection := intersect(options.includeSchemas, options.excludeSchemas); len(intersection) > 0 {
+		return nil, fmt.Errorf("schemas %v are both included and excluded", intersection)
+	}
+
+	includeFilter := buildIncludeSchemasFilter(options.includeSchemas)
+	excludeFilter := buildExcludeSchemasFilter(options.excludeSchemas)
+	return andNameFilter(includeFilter, excludeFilter), nil
```
Comment (bplunkett-stripe) [inline internal/schema/schema.go:414]: We technically don't need to build a `excludeSchemasFilter` if there is an `includeSchemasFilter`, however I think the flow of control here is a bit simpler.

Everything will also still work if we got rid of the intersection check with this method!

File: pkg/tempdb/factory.go [hunk 2]
```diff
@@ -185,12 +202,18 @@ func (o *onInstanceFactory) Create(ctx context.Context) (sql *sql.DB, dropper Dr
...
+		return nil, err
 	}
 
-	return tempDbConn, func(ctx context.Context) error {
-		_ = tempDbConn.Close()
-		return o.dropTempDatabase(ctx, tempDbName)
+	return &Database{
+		ConnPool: tempDbConn,
+		ExcludeMetadatOptions: []schema.GetSchemaOptions{
+			schema.WithExcludeSchemas(o.options.metadataSchema),
```
Comment (bplunkett-stripe) [inline pkg/tempdb/factory.go:211]: This is the core of the PR. Without this, all acceptance tests would fail because they would try to diff out the metadata table.
